store/proxy/proxy_store.go

```diff
@@ -568,3 +578,40 @@ func (s *Store) fromInternal(apiContext *types.APIContext, schema *types.Schema,
 
 	return data
 }
+
+// getListStruct returns a runtime object for storing results from list requests.  If the Store's scheme does not return
+// a type for the resource associated with the store, a generic type will be used.
+func (s *Store) getListStruct() runtime.Object {
+	// try to find the list type for this store
+	obj, err := s.typer.New(schema.GroupVersionKind{
+		Group:   s.group,
+		Version: s.version,
+		Kind:    s.kind + "List",
+	})
+
+	// if we cannot get the specific type default to a generic parser
+	if err != nil {
+		logrus.Debugf("Falling back to generic list type for [%s]", s.kind)
+		return new(unstructured.UnstructuredList)
+	}
+
+	return obj
+}
+
+// collectionFromInternal maps a collection runtime object to an array of maps.
+func (s *Store) collectionFromInternal(list runtime.Object, apiContext *types.APIContext, schema *types.Schema) []map[string]interface{} {
+	var ul unstructured.UnstructuredList
+
+	// Since we are converting to the generic version of a collection any collection type will be convertable.  If a non
+	//collection type is passed to this method the line below will return an error and the method will return an empty
+	//result set.  There is no way to handle this error returned because it is only returned if this method is used
+	//improperly. We swallow the error here to prevent having to check it in the caller.
+	_ = s.typer.Convert(list, &ul, nil)
```

Comment:
Ultimately, it seems that the explanation that "if this produces an error, then you're using it incorrectly" is not a good reason to not return an error.

In the unlikely case that this does produce an error, it would be great to have it returned or at least logged to aid in debugging.

I'd be happy to discuss this offline, if necessary.

Reply:
we discussed this in standup.  We will add a log message here so that in the unlikely event an error occurs we can find it.